Neos.Flow/Classes/Security/Account.php

```diff
@@ -177,15 +187,23 @@ public function setAuthenticationProviderName($authenticationProviderName)
         $this->authenticationProviderName = $authenticationProviderName;
     }
 
+    /**
+     * @return AuthenticationProviderName
+     */
+    public function getProviderName(): AuthenticationProviderName
```

Comment:
Wait, why didn't you change `getAuthenticationProviderName` but introduced a new method? `getAuthenticationProviderName` is used a lot in the authentication code

Reply:
A matter of B/C mentioned in https://discuss.neos.io/t/could-we-build-a-version-of-flow-without-doctrine-orm-database-requirement/2187/17?u=sorenmalling by @bwaidelich Try and avoid breaking more than nescessary and maybe deprecate over time.

Reply:
I could replace it with the return of a object and everywhere in the core, cast it to a string. And write a !!! about it for the commit message, so everyone building for Flow 6 can adjust there code?

Reply:
I would prefer a breaking change now over having two methods that return the same(ish) thing and also having the new one being less descriptive than the old. WDYT @bwaidelich ?

Reply:
"BC" edition just comitted :)

Reply:
We have the same case with getIdentifier / getAccountIdentifier - i will replace and adjust that tomorrow

Reply:
Just created a new commit that uses the AccountIdentifier object as return value